### DISCUSSION TITLE
why do I have to enter github username and password everytime I push?

### DISCUSSION THREAD

**Discussion by LakinduK** [community/community#21768] [New to GitHub]
<p>Im using github in linux (Pop OS 19.10). Everytime I try to push, Im asked for github username &amp; password.</p>
<p>Is there any method to save it forever?</p>

**Comment by logankilpatrick** [comment, marked as answer]
<p>Hey <a class="mention" href="/lakinduk">@lakinduk</a>, thanks for being here! See here: <a href="https://stackoverflow.com/questions/35942754/how-to-save-username-and-password-in-git" rel="nofollow noopener">https://stackoverflow.com/questions/35942754/how-to-save-username-and-password-in-git</a> for details on saving your credentials though the more secure method would be to use an SSH key as detailed here: <a href="https://help.github.com/en/github/authenticating-to-github/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent" rel="nofollow noopener">https://help.github.com/en/github/authenticating-to-github/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent</a></p>
